plugin/trino-hive/src/main/java/io/trino/plugin/hive/CachingDirectoryLister.java

```diff
@@ -106,7 +190,13 @@ public boolean hasNext()
             {
                 boolean hasNext = iterator.hasNext();
                 if (!hasNext) {
-                    cache.put(path, ImmutableList.copyOf(files));
+                    try {
+                        cache.get(path, () -> ImmutableList.copyOf(files));
```

Comment:
This workarounds the fact that `cache.put` is disabled (for a reason). However, the flow is no different that it used to be. While it effectively circumvents the safety check in `cache.put` implementation, it's not a safe operation.

Reply:
I fail to see when the cache would return a stale value in the context of the `CachingDirectoryLister`.
Note that this is the only operation adding values to the cache.